modules/dataquery/static/schema.yml

```diff
@@ -0,0 +1,331 @@
+          schema:
+            type: string
+            enum: ['top', 'untop', 'dashboard']
+        - name: name
+          in: query
+          style: pipeDelimited
+          schema:
+            type: string
+      responses:
+        '400':
+          description: Bad request. Most likely caused by one or more incompatible patches to be done at the same time.
+  /queries/{QueryID}/run:
+    parameters:
+      - name: QueryID
+        in: path
+        description: the QueryID returned by posting to /queries
+        required: true
+        style: simple
+        schema:
+          type: integer  
+    get:
+      description: |- 
+        Run the query QueryID and returns the results. 
+        
+        This endpoint will result in a new query run being generated, which will be returned in the 'recent' queries of the user on the /queries endpoint.
+      responses:
+        '200':
+          description: The query was able to be successfully run
+          content:
+            application/octet-stream:
+              schema:
+                $ref: '#/components/schemas/QueryResults'              
+        '500':
+          description: Something went wrong on the server running the query
+  /queries/{QueryID}/count:      
+    parameters:
+    - name: QueryID
+      in: path
+      description: the QueryID returned by posting to /queries
+      required: true
+      style: simple
+      schema:
+        type: integer  
+    get:
+      responses:
+        '200':
+          description: |-
+           A count of the number of candidate matches that would be returned if the query were to be run by the current user right now.
+           
+           This endpoint does *not* result in a new query run being generated or run the query, it only returns the count of how many candidates would match if the query *were* to be run.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  count:
+                    type: integer
+                    description: The number of candidates that match the filters 
+                    example: 34
+  /queries/{QueryID}/run/{QueryRunID}: 
+    description: |-
+      Returns the cached results of a previously run query
+      
+      Note: This endpoint is aspirational.
+    parameters:
+    - name: QueryID
+      in: path
+      description: the QueryID returned by posting to /queries
+      required: true
+      style: simple
+      schema:
+        type: integer
+    - name: QueryRunID
+      in: path
+      description: the identifier of a previous run for this QueryID.
+      required: true
+      style: simple
+      schema:
+        type: integer 
+    get:
+      responses:
+        '500':
+          description: Not implemented
+          content:
+            application/json:
+              schema:
+                type: object
+            
+components:
+  schemas:
+    AllQueries:
```

Comment:
Not sure why there are 3 properties here.
They can be known by checking the `Stared` and `Shared` properties of each queries. I would add `Top` or `Pinned` for the third one.

Reply:
You are probably right. I'll have to look into the backend code to see if this is just because of how the code evolved or if there's some reason related to efficiency that I did this.